### ISSUE TITLE
🐛 Bug: Flat config causes eslint-plugin-import error message

### ISSUE THREAD

**Issue by KristjanESPERANTO** [JoshuaKGoldberg/eslint-plugin-package-json#247]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

I'm using the recommended rules of [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import) also for checking `eslint.config.js`. I just added `eslint-plugin-package-json` according to the README instructions to the config like this:

```js
import eslintPluginImport from "eslint-plugin-import";
import packageJson from "eslint-plugin-package-json/configs/recommended"; // added
```

I wouldn't expect any complaints from the import plugin when I import `eslint-plugin-package-json`.

### Actual

Now when I run eslint I get the import rule [no-unresolved](https://github.com/import-js/eslint-plugin-import/blob/v2.29.1/docs/rules/no-unresolved.md) throws that error message:

```
error    Unable to resolve path to module 'eslint-plugin-package-json/configs/recommended'  import/no-unresolved
```

`eslint-plugin-package-json`  works nevertheless.

I have no other problems with the import plugin, so I guess there is a problem with the way eslint-plugin-package-json is imported.

**Comment by JoshuaKGoldberg** [comment]
Hmm, this might be https://github.com/import-js/eslint-plugin-import/issues/2495. Have you tried with [`eslint-plugin-import-x`](https://github.com/un-ts/eslint-plugin-import-x)?

**Comment by KristjanESPERANTO** [comment]
I didn't know `eslint-plugin-import-x` before, but I just tested it. Saddly the same problem appears:

```
error    Unable to resolve path to module 'eslint-plugin-package-json/configs/recommended'  import-x/no-unresolved
```

It seems, that the issue is the same in eslint-plugin-import and eslint-plugin-import-x and not in your package. So we can close this issue?

I'll deactivate the rule for now.

**Comment by JoshuaKGoldberg** [comment]
Yeah agreed. It's unfortunate but given that the path resolution does work at runtime, I'm not inclined to do a deep dive into other plugins' issues.

Thanks for filing though! If someone else sees this then, hey, maybe they'll have info to support or dispute this conclusion. 😄
